Replace serfmembership with k8smembership

bernardjkim opened the pull request:
## Description
This PR replaces the cluster membership implementation with one that relies directly on Kubernetes. Satellite no longer relies on a Serf cluster to query cluster membership, instead all Nodes in the Kubernetes cluster will be considered a member of the cluster.

Will work on completely removing Serf packages and references in a later PR.

## Linked tickets and PRs
* Requires https://github.com/gravitational/satellite/pull/282

bernardjkim committed:
Revendor

bernardjkim committed:
Replace serfmembership with k8smembership

a-palchikov commented on lib/membership/kubernetes/kubernetes_test.go:
```diff
@@ -0,0 +1,137 @@
+		c.Assert(err, IsNil, comment)
+
+		members, err := cluster.Members()
+		sort.Sort(pb.ByName(members))
```
Any reason why you sort the result before checking the error?

bernardjkim replied:
No reason. Will change to call sort after checking the error.

a-palchikov commented on lib/membership/kubernetes/kubernetes_test.go:
```diff
@@ -0,0 +1,137 @@
+
+		stop := make(chan struct{})
+		defer close(stop)
+		go informer.Run(stop)
```
nit: this is going to get called only upon exit from the test method - have you considered confining the actual test in a separate function. In case more test cases are added, this might have a side-effect.
On the other hand, it looks like there's just a single test-case - why use table-driven testing prematurely?

bernardjkim replied:
Good point. I guess there is no need to use table-driven testing here.

a-palchikov commented on lib/membership/kubernetes/kubernetes.go:
```diff
@@ -0,0 +1,135 @@
+	Informer cache.SharedIndexInformer
+	// Stop specifies a stop channel that indicates that the informer has been
+	// stopped when it is closed.
+	Stop <-chan struct{}
```
Is this used for anything?

bernardjkim committed:
Add publicip tag to be read by Gravity

bernardjkim committed:
Address feedback

- Sort after checking error
- Remove premature use of table-driven testing
- Remove unused Stop field
